pkg/model/manifest.go

```diff
@@ -663,19 +663,21 @@ func Read(bytes []byte) (*Manifest, error) {
 
 	hasShownWarning := false
 	for _, dev := range manifest.Dev {
-		if (dev.Image.Context != "" || dev.Image.Dockerfile != "") && !hasShownWarning {
+		if dev.Image != nil && (dev.Image.Context != "" || dev.Image.Dockerfile != "") && !hasShownWarning {
 			hasShownWarning = true
 			oktetoLog.Yellow(`The 'image' extended syntax is deprecated and will be removed in a future version. Define the images you want to build in the 'build' section of your manifest. More info at https://www.okteto.com/docs/reference/manifest/#build"`)
 		}
+
 	}
 
-	if err := manifest.setDefaults(); err != nil {
```

Comment:
we are setting default values before validating, this is not very useful because if the manifest is invalid, we should fail early and make the `setDefaults` code safer, because right now it accesses properties that might not be there, which are caught by the `validate`.

Reply:
I think some validation depends on the values configured in the `setDefaults` func for example `divert.driver`

Reply:
@AdrianPedriza I had a look but I can't find a logic that could be an issue with this change. If you have any more specific suggestion, I am more than happy to look at it 🙏

Reply:
[here](https://github.com/okteto/okteto/blob/master/pkg/model/manifest.go#L789) we have a switch that check divert driver and that values can be comnfigured in `setDefault` [here](https://github.com/okteto/okteto/blob/master/pkg/model/manifest.go#L836) I think

Reply:
Good catch @AdrianPedriza 👏 

I've reverted the order and added an additional `if` to skip when build is nil. Check it out in this commit: ead42ab9